packages/context/src/context.ts

```diff
@@ -197,22 +192,49 @@ export class Context {
    *
    * @param keyWithPath The binding key, optionally suffixed with a path to the
    *   (deeply) nested property to retrieve.
+   * @returns A promise of the bound value.
+   */
+  get<T>(keyWithPath: string): Promise<T>;
+
+  /**
+   * Get the value bound to the given key, optionally return a (deep) property
+   * of the bound value.
+   *
+   * @example
+   *
+   * ```ts
+   * // get "rest" property from the value bound to "config"
+   * // use "undefined" when not config was provided
+   * const config = await ctx.get('config#rest', {optional: true});
```

Comment:
Shall we modify the code snippet in this example as well? `config = await ctx.get<RestComponentConfig>('config#rest', {optional: true});` (see https://github.com/strongloop/loopback-next/blob/master/packages/rest/src/rest-component.ts#L48)

Reply:
Good catch! I'll edit the snippet.